starclass/RFGCClassifier/RF_GC_featcalc.py

```diff
@@ -502,12 +525,15 @@ def freq_phasediffs(featdictrow, usedfreqs):
 		phase difference of 2nd to 1st and 3rd to 1st frequencies
 
 	"""
-	if len(usedfreqs) >= 2:
-		phi21 = featdictrow['phase'+str(usedfreqs[1]+1)] - 2*featdictrow['phase'+str(usedfreqs[0]+1)]
+	tab=featdictrow['frequencies']
+	if usedfreqs >= 2:
+		#phi21 = featdictrow['phase'+str(usedfreqs[1]+1)] - 2*featdictrow['phase'+str(usedfreqs[0]+1)]
+		phi21 = tab[(tab['num'] == 2) & (tab['harmonic']== 0)]['phase'] - 2*tab[(tab['num'] == 1) & (tab['harmonic']== 0)]['phase']
```

Comment:
We are doing the same slicing several times... Maybe just do it once like
```
peak1 = tab[(tab['num'] == 1) & (tab['harmonic']== 0)]
peak2 = tab[(tab['num'] == 2) & (tab['harmonic']== 0)]
amp21 = peak2['amplitude'] / peak1['amplitude']
phi21 = peak2['phase'] - 3*peak1['phase']
```
and so on...

Reply:
Added it for peak1 in freq_ampratios & freq_phasediffs as the other peaks are used only once.